src/models/job.py

```diff
@@ -29,85 +33,127 @@ class Job(Base):
+                symbol = 'Pass'
+            else:
+                symbol = 'Failed: {}'.format(self.exit_code)
+            args = [self.id, self.node, symbol]
+            click.echo('ID: {}, Node: {}, {}'.format(*args))
+
+        async def __run_thread(self, func, *a):
+            def catch_errors(func, *args):
+                try: func(*args)
+                except: pass
+            run = lambda: catch_errors(func, *a)
+            coroutine = self._loop.run_in_executor(self.thread_pool, run)
+            return await coroutine
+
+        async def run_async(self):
+            if self.check_command():
+                try: await self.__run_thread(self.connection().open)
+                except concurrent.futures.CancelledError as e: raise e
+
+                if self.connection().is_connected:
+                    await self.__run_thread(self.run, self.batch)
+                else:
+                    self.set_ssh_error()
+
+    def task(self, *a, **k): return Job.JobTask(self, *a, **k)
+
+    def check_command(self):
+        if self.batch.command_exists(): return True
+        else:
+            self.stdout = ''
+            self.stderr = 'Incorrectly configured command'
+            self.exit_code = -2
+
+    def set_ssh_error(self):
+        self.stdout = ''
+        self.stderr = 'Could not establish ssh connection'
+        self.exit_code = -1
+
+    def set_ssh_results(self):
+        if self.__result == None: return
+        if self.batch.is_interactive():
+            self.stdout = 'Interactive Job: STDOUT is unavailable'
+            self.stderr = 'Interactive Job: STDERR is unavailable'
+            self.exit_code = -3
+        else:
+            self.stdout = self.__result.stdout
+            self.stderr = self.__result.stderr
+            self.exit_code = self.__result.return_code
+
+    def run(self, batch):
+        def __with_tempdir(func):
+            def wrapper(*args):
+                result = self.connection().run('mktemp -d', hide='both')
+                if result:
+                    temp_dir = ('{}'.format(result.stdout)).rstrip()
                     try:
-                        func(connection)
+                        result = func(temp_dir, *args)
                     finally:
-                        connection.close()
+                        self.connection().run("rm -rf {}".format(temp_dir))
+                return result
             return wrapper
 
-        @__check_command
-        @__with_connection
-        def __runner(connection):
-            def __set_result(result):
-                if self.batch.is_interactive():
-                    self.stdout = 'Interactive Job: STDOUT is unavailable'
-                    self.stderr = 'Interactive Job: STDERR is unavailable'
-                    self.exit_code = -3
+        @__with_tempdir
+        def __run_command(temp_dir):
+            # Copies the files across
+            path = '/var/lib/adminware/tools/namespace/stutool1/config.yaml'
```

Comment:
What's this line doing? A leftover from testing?

Reply:
Removed: https://github.com/alces-software/adminware/pull/155/commits/3e85d8682bd2bfb609e2c5db7eadb35ec4768071